### PR TITLE
EPMDJ-5136 Add an action for Baseline

### PR DESCRIPTION
**Pull request by LikhanovMaxim**
stored 'baseline' in DB; init first build as baseline; added action for updating current build as baseline

### CODE REVIEW

File: api/src/commonMain/kotlin/Model.kt [hunk 1]
```diff
@@ -75,6 +75,11 @@ data class RenameScopePayload(
 @Serializable
 data class ScopePayload(val scopeId: String = "")
 
+@Serializable
+data class AgentBaselinePayload(
```
Comment (viktorbll) [inline api/src/commonMain/kotlin/Model.kt:79]:
```suggestion
data class BaselinePayload(
```

File: admin-part/src/main/kotlin/Build.kt [hunk 2]
```diff
@@ -48,3 +49,23 @@ internal fun CoverContext.toBuildStatsDto(): BuildStatsDto = BuildStatsDto(
     unaffected = methodChanges.unaffected.count(),
     deleted = methodChanges.deleted.count()
 )
+
+suspend fun Plugin.updateBaseline(payload: AgentBaselinePayload): ActionResult {
```
Comment (viktorbll) [inline admin-part/src/main/kotlin/Build.kt:53]: move to BuildHandler.kt

File: admin-part/src/main/kotlin/Plugin.kt [hunk 3]
```diff
@@ -297,7 +298,7 @@ class Plugin(
     ) {
         val coverageInfoSet = calculateCoverageData(context)
         val buildMethods = coverageInfoSet.buildMethods
-        val parentVersion = adminData.buildManager[buildVersion]?.parentVersion?.takeIf(String::any)
+        val parentVersion = storeClient.loadBaseline(agentId)?.takeIf(String::any)
```
Comment (viktorbll) [inline admin-part/src/main/kotlin/Plugin.kt:301]: No need for a special method here, use findById.

File: admin-part/src/main/kotlin/State.kt [hunk 4]
```diff
@@ -133,10 +133,13 @@ internal class AgentState(
             build = build
         )
         _coverContext.value = coverContext
-        buildInfo?.parentVersion?.takeIf(String::any)?.let { parentVersion ->
-            storeClient.loadClassData(parentVersion)?.let { parentClassData ->
+        val agentId = agentInfo.id
+        val baselineVersion = storeClient.loadBaseline(agentId)
```
Comment (viktorbll) [inline admin-part/src/main/kotlin/State.kt:137]: parentVersion

File: admin-part/src/main/kotlin/storage/Build.kt [hunk 5]
```diff
@@ -60,3 +60,7 @@ internal suspend fun CachedBuild.store(storage: StoreClient) {
     }
 }
 
+internal suspend fun StoreClient.loadBaseline(
```
Comment (viktorbll) [inline admin-part/src/main/kotlin/storage/Build.kt:63]: remove this

File: api/src/commonMain/kotlin/Model.kt [hunk 6]
```diff
@@ -75,6 +75,11 @@ data class RenameScopePayload(
...
+    val baseline: Boolean
```
Comment (viktorbll) [inline api/src/commonMain/kotlin/Model.kt:80]: isBaseline

**Review by viktorbll — CHANGES_REQUESTED** [review]

File: api/src/commonMain/kotlin/Model.kt [hunk 7]
```diff
@@ -80,6 +80,11 @@ data class ParentBuildDto(
     val version: String
 )
 
+@Serializable
+data class BaselineDto(
+    val version: String
+)
```
Comment (LikhanovMaxim) [inline api/src/commonMain/kotlin/Model.kt:86]: maybe it should union ParentBuildDto & BaselineDto to Dto, for example:

> @Serializable
> data class BuildDto(
>     val version: String
> )

Comment (viktorbll) [reply api/src/commonMain/kotlin/Model.kt:86]: BuildVersionDto

**Review by viktorbll — APPROVED** [review]

**Pull request merged**